app/src/main/java/ch/epfl/skysync/components/ChatText.kt

```diff
@@ -0,0 +1,204 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import ch.epfl.skysync.ui.theme.lightOrange
+
+data class ChatMessage(
+    val sender: String,
+    val profilePicture: ImageVector?,
+    val message: String,
+    val time: String
+)
+
+/**
+ * Composable function representing a chat screen.
+ *
+ * @param groupName Name of the group or chat.
+ * @param msgList List of chat messages with sender information, message content, and timestamp.
+ * @param backClick Callback function to be invoked when the back button is clicked.
+ * @param sendClick Callback function to be invoked when a message is sent.
+ * @param paddingValues Padding values to be applied to the chat screen.
+ */
+@Composable
+fun ChatText(
+    groupName: String,
+    msgList: List<ChatMessage>,
+    backClick: () -> Unit,
+    sendClick: (String) -> Unit,
+    paddingValues: PaddingValues
+) {
+  Column(modifier = Modifier.fillMaxSize().padding(paddingValues)) {
+    Header(BackClick = backClick, title = groupName)
+    Spacer(modifier = Modifier.size(1.dp))
+    ChatTextBody(msgList)
+    ChatInput(sendClick)
+  }
+}
+/**
+ * Composable function representing the body of a chat screen, displaying chat bubbles.
+ *
+ * @param msgList List of chat messages with sender name, sender profile picture, message content,
+ *   and timestamp.
+ */
+@Composable
+fun ChatTextBody(msgList: List<ChatMessage>) {
+  val lazyListState = rememberLazyListState()
+  LazyColumn(Modifier.fillMaxHeight(0.875f).testTag("ChatTextBody"), state = lazyListState) {
+    items(msgList.size) { index ->
+      ChatBubble(
+          sender = msgList[index].sender,
+          image = msgList[index].profilePicture,
+          message = msgList[index].message,
+          time = msgList[index].time,
+          index = "$index")
+    }
+  }
+  LaunchedEffect(Unit) { lazyListState.scrollToItem(msgList.size - 1) }
+}
+/**
+ * Composable function representing a chat bubble.
+ *
+ * @param sender The sender of the message.
+ * @param image The image associated with the sender.
+ * @param message The message content.
+ * @param time The timestamp of the message.
+ * @param index The index of the chat bubble.
+ */
+@Composable
+fun ChatBubble(sender: String, image: ImageVector?, message: String, time: String, index: String) {
+  var isMyMessage = false
+  if (sender == "me") {
+    isMyMessage = true
```

Comment:
Instead of checking for some specific value of `sender`, consider adding a attribute message type to `ChatMessage`:
```kotlin
enum class MessageType {
    SENT,
    RECEIVED,
}

data class ChatMessage(
    val sender: String,
    val messageType: MessageType,
    val profilePicture: ImageVector?,
    val message: String,
    val time: String
)
```

Reply:
Thank you didn't think of this solution, going to change it